src/celeritas/user/detail/StepGatherAction.cc

```diff
@@ -59,6 +59,10 @@ void StepGatherAction<P>::execute(CoreHostRef const& core) const
 {
     CELER_EXPECT(core);
 
+    // Lock mutex to prevent multiple CPU threads from
+    // creating/accessing/processing state data simultaneously
+    std::lock_guard<std::mutex> scoped_lock{storage_->mumu};
```

Comment:
is storage_ shared by threads?  Why isn't it thread local, then merged at the end of run?

Reply:
Yeah, and the other thing is that `thread_local` only works with global data, which would limit us to a single  `StepCollector`. As discussed earlier we'll probably want to have more than one in the future.